webapp/blog/views.py

```diff
@@ -1,225 +1,32 @@
-        return flask.abort(503)
-    except ApiError:
-        return flask.abort(502)
+        blog_articles = None
+        articles = []
 
-    right_urls = logic.change_url(
-        feed, flask.request.base_url.replace("/feed", "")
-    )
-
-    right_title = right_urls.replace("Ubuntu Blog", "Snapcraft Blog")
-
-    return flask.Response(right_title, mimetype="text/xml")
-
-
-@blog.route(
-    '/<regex("[0-9]{4}"):year>/<regex("[0-9]{2}"):month>/'
-    '<regex("[0-9]{2}"):day>/<slug>'
-)
-@blog.route('/<regex("[0-9]{4}"):year>/<regex("[0-9]{2}"):month>/<slug>')
-@blog.route('/<regex("[0-9]{4}"):year>/<slug>')
-def article_redirect(slug, year, month=None, day=None):
-    return flask.redirect(flask.url_for(".article", slug=slug))
-
-
-@blog.route("/<slug>")
-def article(slug):
-    try:
-        articles = api.get_article(slug)
-    except ApiCircuitBreaker:
-        return flask.abort(503)
-    except ApiError as api_error:
-        return flask.abort(502, str(api_error))
-
-    if not articles:
-        flask.abort(404, "Article not found")
-
-    article = articles[0]
-
-    try:
-        author = article["_embedded"]["author"][0]
-    except (IndexError, KeyError):
-        author = None
-
-    transformed_article = logic.transform_article(
-        article, author=author, optimise_images=True
-    )
-
-    try:
-        # Tags live in index 1 of wp:term
-        tags = article["_embedded"]["wp:term"][1]
-        tags = [{"id": tag["id"], "name": tag["name"]} for tag in tags]
-    except (IndexError, KeyError):
-        tags = []
-
-    try:
-        related_articles, total_pages = api.get_articles(
-            tags=tags, per_page=3, exclude=article["id"]
-        )
-    except ApiError:
-        related_articles = None
-
-    if related_articles:
-        for related_article in related_articles:
-            related_article = logic.transform_article(related_article)
-
-    newsletter_subscribed = flask.request.args.get(
-        "newsletter", default=False, type=bool
-    )
-
-    context = {
-        "article": transformed_article,
-        "related_articles": related_articles,
-        "tags": tags,
-        "is_in_series": logic.is_in_series(tags),
-        "newsletter_subscribed": newsletter_subscribed,
-    }
-
-    return flask.render_template("blog/article.html", **context)
-
-
-@blog.route("/api/snap-posts/<snap>")
-def snap_posts(snap):
-    try:
-        blog_tags = api.get_tag_by_name("".join(["sc:snap:", snap]))
-    except ApiError:
-        blog_tags = None
-
-    blog_articles = None
-    articles = []
-
-    if blog_tags:
-        blog_tags_ids = logic.get_tag_id_list(blog_tags, snap)
-
-        if blog_tags_ids:
+        if blog_tags:
             try:
-                blog_articles, total_pages = api.get_articles(blog_tags_ids, 3)
+                blog_articles, total_pages = wordpress_api.get_articles(
+                    blog_tags["id"], 3
+                )
             except ApiError:
```

Comment:
I think ApiError won't work here, since `wordpress_api` does not raise this exception.